crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1059,6 +1059,43 @@ impl IndexerStore for PgIndexerStore {
         self.multi_get_transactions_by_digests(&tx_digests).await
     }
 
+    async fn get_transaction_page_by_address(
+        &self,
+        address: SuiAddress,
+        start_sequence: Option<i64>,
+        limit: usize,
+        is_descending: bool,
+    ) -> Result<Vec<Transaction>, IndexerError> {
+        let sql_query = format!(
+            "SELECT transaction_digest as digest_name FROM (
+                SELECT transaction_digest, max(id) AS max_id
+                FROM recipients
+                WHERE recipient = '{}' OR sender = '{}'
+                {} GROUP BY transaction_digest
+                ORDER BY max_id {} LIMIT {}
+            ) AS t",
+            address,
+            address,
+            if let Some(start_sequence) = start_sequence {
+                if is_descending {
+                    format!("AND id < {}", start_sequence)
+                } else {
+                    format!("AND id > {}", start_sequence)
+                }
+            } else {
+                "".to_string()
+            },
+            if is_descending { "DESC" } else { "ASC" },
+            limit
+        );
+        let tx_digests: Vec<String> = read_only_blocking!(&self.blocking_cp, |conn| diesel::sql_query(sql_query).load(conn))
```

Comment:
apparently I don't know much about the indexer code - how do we prevent SQL injection here? is it handled by diesel?

Reply:
this is read-only so no data will commit from the query; also the underlying DB replica is read-only too.

Reply:
there could be sql injection in read mode too. Maybe we can comfort ourselves that all of these are public data so we are not afraid but still it's concerning we are doing raw sql queries

Reply:
yeah indeed, it's all public info here so it would not hurt to read them.
Also the input argument / parameter is a SuiAddress, so clients will not be able to draft their own arbitrary query either.